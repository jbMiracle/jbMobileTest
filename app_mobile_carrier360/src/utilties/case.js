
export const camelToUnderscore = text => text.replace(/([a-z])([A-Z])/g, (match, one, two) => `${one}_${two}`);

export const camelToSnake = text => camelToUnderscore(text).toLowerCase();
