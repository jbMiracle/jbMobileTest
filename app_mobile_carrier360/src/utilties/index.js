export * from './case';
export * from './validators';
export * from './formatters';
