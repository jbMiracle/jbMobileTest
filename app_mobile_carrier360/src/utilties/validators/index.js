export * from './emailValidator/emailValidator';
export * from './phoneNumberValidator/phoneNumberValidator';
