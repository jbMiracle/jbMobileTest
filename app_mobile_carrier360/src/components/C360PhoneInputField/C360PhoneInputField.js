import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { omit } from 'lodash';
import { C360InputField } from '../C360InputField/C360InputField';
import { isPhoneNumberValid, formatPhoneNumber } from '../../utilties';

class C360PhoneInputField extends PureComponent {
  validatePhone = (phoneNumber) => {
    const { onValidationUpdate, onChangeText } = this.props;
    const formattedNumber = formatPhoneNumber(phoneNumber);
    const hasValidPhoneNumber = isPhoneNumberValid(phoneNumber);

    if (onChangeText) {
      onChangeText(formattedNumber);
    }

    if (onValidationUpdate) {
      onValidationUpdate(hasValidPhoneNumber);
    }
  };

  render() {
    const options = omit(this.props, ['onValidationUpdate', 'onChangeText', 'keyboardType']);

    return (
      <C360InputField
        { ...options }
        keyboardType='phone-pad'
        maxLength={ 14 }
        onChangeText={ this.validatePhone }
      />
    );
  }
}

C360PhoneInputField.propTypes = {
  onValidationUpdate: PropTypes.func,
  onChangeText: PropTypes.func,
};

C360PhoneInputField.defaultProps = {
  onValidationUpdate: undefined,
  onChangeText: undefined,
};

export default C360PhoneInputField;
