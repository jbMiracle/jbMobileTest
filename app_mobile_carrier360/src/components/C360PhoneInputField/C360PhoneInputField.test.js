import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import C360PhoneInputField from './C360PhoneInputField';
import { isPhoneNumberValid, formatPhoneNumber } from '../../utilties';

jest.mock('../../utilties', () => ({
  isPhoneNumberValid: jest.fn(),
  formatPhoneNumber: jest.fn(),
}));

const props = {
  onValidationUpdate: jest.fn(),
  onChangeText: jest.fn(),
};

describe('C360PhoneInputField', () => {
  describe('performs the validatePhone operation in a pretend context', () => {
    isPhoneNumberValid.mockImplementation(() => 'abc');
    formatPhoneNumber.mockImplementation(() => 'xyz');
    const phoneField = new C360PhoneInputField(props);
    phoneField.validatePhone('some text');

    it('should validate the inputted text and call onValidationUpdate with the result', () => {
      expect(props.onValidationUpdate.mock.calls[0][0]).toBe('abc');
    });

    it('should format the inputted text and call onChangeText with the result', () => {
      expect(props.onChangeText.mock.calls[0][0]).toBe('xyz');
    });
  });

  it('can render an empty component', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<C360PhoneInputField />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
