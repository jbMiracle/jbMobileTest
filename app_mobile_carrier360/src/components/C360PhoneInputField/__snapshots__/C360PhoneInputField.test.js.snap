// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`C360PhoneInputField can render an empty component 1`] = `
<withAnimatable(Styled(C360InputField))
  animation={undefined}
  delay={0}
  direction="normal"
  duration={undefined}
  easing={undefined}
  iterationCount={1}
  keyboardType="phone-pad"
  maxLength={14}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onChangeText={[Function]}
  style={undefined}
  transition={undefined}
  useNativeDriver={false}
/>
`;
